fili-core/src/main/java/com/yahoo/bard/webservice/data/config/table/BaseTableLoader.java

```diff
@@ -57,33 +60,23 @@ protected BaseTableLoader() {
         this(DateTimeZone.UTC);
     }
 
-    @Override
-    public abstract void loadTableDictionary(ResourceDictionaries dictionaries);
-
     /**
-     * Builds a table group.
-     * <p>
-     * Builds and loads the physical tables for the physical table definitions as well.
-     *
-     * @param logicalTableName  The logical table for the table group
-     * @param apiMetrics  The set of metric names surfaced to the api
-     * @param druidMetrics  Names of druid datasource metric columns
-     * @param tableDefinitions  A list of config objects for physical tables
-     * @param dictionaries  The container for all the data dictionaries
+     * Load user configured tables into resource dictionary.
      *
-     * @return A table group binding all the tables for a logical table view together.
+     * @param dictionaries dictionary to be loaded with user configured tables
      *
-     * @deprecated logicalTableName is not used in TableGroup
+     * @deprecated should use the same method that takes an additional DataSourceMetadataService argument instead
      */
     @Deprecated
-    public TableGroup buildTableGroup(
-            String logicalTableName,
-            Set<ApiMetricName> apiMetrics,
-            Set<FieldName> druidMetrics,
-            Set<PhysicalTableDefinition> tableDefinitions,
-            ResourceDictionaries dictionaries
+    public abstract void loadTableDictionary(ResourceDictionaries dictionaries);
+
+    @Override
+    public void loadTableDictionary(
+            ResourceDictionaries dictionaries,
+            DataSourceMetadataService metadataService
     ) {
-        return buildTableGroup(apiMetrics, druidMetrics, tableDefinitions, dictionaries);
+        this.metadataService = metadataService;
+        this.loadTableDictionary(dictionaries);
```

Comment:
This setup is awkward.

We've deprecated the old version, which doesn't take the `metadataService`. This implies that we shouldn't be using the old method at all. Meanwhile, this version of the function just assigns the class level `metadataService` and then calls into the old version. That's fine as far as preserving backwards compatibility. 

However, near as  I can tell, the "bridge to preserve backwards compatibility" is now the new version of this class is _intended_ to be used. People are meant to override `loadTableDictionary`, and then call into the various helper functions to do the heavy lifting for them. However, none of those helper functions take the `metadataService` as an argument, and instead rely on the `metadataService` field.

So in other words, if people want to do the "right" thing, and stop using the deprecated method, and they want to do the "right" thing and use the various helper methods, they have to:

1. Set a field in the super class.
2. Call their old code and let magic happen.

Instead, I think we should deprecate _all_ the helper methods that rely on `metadataService` but don't take an argument (`loadPhysicalTable`, `buildPhysicalTable`, `buildTableGroup`), in favor of a version that does take the `metadataService` as an argument. It will be intended then, that people will call the helper methods that take the `metadataService` as an argument, and pass it through to them from their version of `loadTableDictionary`.

The deprecated versions can then call into the new versions, passing in the `segmentMetadata` field.

That way, the blessed way of using the `BaseTableLoader` doesn't have to do any awkward superclass field manipulation, and we still have the same backwards compatibility as we do now.

@michael-mclawhorn Does this make sense to you?